packages/web-app-files/src/components/SideBar/Shares/Collaborators/InviteCollaborator/InviteCollaboratorForm.vue

```diff
@@ -321,24 +319,46 @@ export default defineComponent({
             }
 
             const addMethod = this.resourceIsSpace ? this.addSpaceMember : this.addShare
-            addMethod({
-              ...this.$language,
-              client: this.$client,
-              graphClient: this.clientService.graphAuthenticated,
-              path,
-              shareWith: collaborator.value.shareWith,
-              displayName: collaborator.label,
-              shareType: collaborator.value.shareType,
-              permissions: bitmask,
-              role: this.selectedRole,
-              expirationDate: this.expirationDate,
-              storageId: this.resource.fileId || this.resource.id
-            })
+
+            try {
+              await addMethod({
+                client: this.$client,
+                graphClient: this.clientService.graphAuthenticated,
+                path,
+                shareWith: collaborator.value.shareWith,
+                displayName: collaborator.label,
+                shareType: collaborator.value.shareType,
+                permissions: bitmask,
+                role: this.selectedRole,
+                expirationDate: this.expirationDate,
+                storageId: this.resource.fileId || this.resource.id
+              })
+            } catch (e) {
+              console.error(e)
+              errors.push({
+                displayName: collaborator.label,
+                error: e
+              })
+              throw e
+            }
           })
         )
       })
 
-      await Promise.all(savePromises)
+      const results = await Promise.allSettled(savePromises)
+
+      if (results.find((r) => r.status === 'fulfilled')) {
```

Comment:
```suggestion
      if (results.length !== errors.length) {
```

should work as well right? should be more efficient since you don't need to iterate over the results.